### ISSUE TITLE
PBR roadmap

### ISSUE THREAD

**Issue by lilleyse** [AnalyticalGraphicsInc/cesium#5423]
* [x] Proof-of-concept - Load glTF 1.0 models with hard-coded materials and shaders from https://github.com/khronosGroup/gltf-webgl-pbr. Use the sun as the sole light source.
* [x] Finish up glTF 2.0 support
  * [x] https://github.com/AnalyticalGraphicsInc/cesium/pull/4808
  * [x] https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/272
  * [x] https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/274 - the material translation here is good for testing, but will be removed once PBR support is complete.
* [x] Support normal mapping
  * [x] glTF without `TANGENT` attribute - compute tangent with screen space derivatives if supported
  * [x] glTF with `TANGENT` attribute - just compute bitangent easily
* [x] Lighting ideas
  * [ ] Simple ambient factor
  * [ ] Experiment with default environment maps shipped with Cesium
  * [x] Get atmosphere color / ground color
  * [ ] Render cube map dynamically 
* [ ] Handle [spec-gloss extension](https://github.com/KhronosGroup/glTF/tree/master/extensions/Khronos/KHR_materials_pbrSpecularGlossiness).

@moneimne feel free to add ideas/comments as we go through this. The main first goal is to render with PBR shaders but use the sun as the only light source. After that we can experiment with different light sources like environment maps, atmosphere color, area lights, etc. I'll also add more ideas as we go.

**Comment by pjcozzi** [comment]
CC #4009

**Comment by pjcozzi** [comment]
Before exploring **Lighting ideas** and **spec-gloss extension**, please do the least amount reasonable to get a full glTF 2.0 implementation working - including morph targets - and merged into master.

**Comment by moneimne** [comment]
Idea for applying PBR to geospatial: Altering the metallic and roughness of a model at runtime to achieve certain results, For example, one can add highlights or rim lights to a model to represent selection by manipulating the metallic and roughness parameters.

**Comment by pjcozzi** [comment]
@lilleyse OK to close this?  Is there anything left in the roadmap that should perhaps move to #4009?

**Comment by lilleyse** [comment]
The only thing left unfinished here is `Handle spec-gloss extension.` so good to close.
